hack/verify-file-sizes.sh

```diff
@@ -0,0 +1,80 @@
+#!/usr/bin/env bash
+# Copyright 2023 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+KUBE_ROOT=$(dirname "${BASH_SOURCE[0]}")/..
+source "${KUBE_ROOT}/hack/lib/init.sh"
+cd "${KUBE_ROOT}"
+
+maxsize=$((10 * 1024 * 1024))
+ignorefile=.ignorefilesize
+result=0
+
+# Takes a single file name as parameter and checks whether the ignorefile
+# excludes it. Using git-check-ignore and a file which contains entries
+# like .gitignore would be nicer, but git-check-ignore only looks at
+# .gitignore.
+isignored () {
```

Comment:
do we need this? I can't think of a good reason we'd need to commit MB-sized binary files

Reply:
We did in the past, so perhaps there are legitimate reasons?

I can lower to 1MB and add a local exclusion mechanism in the script itself instead of allowing code owners to add their own `.ignorefilesizes` - then the barrier for adding large files is higher, but it remains possible.

Agreed?

Reply:
> We did in the past, so perhaps there are legitimate reasons?

I'm 99% sure whoever committed that did not know it was so big.

> I can lower to 1MB and add a local exclusion mechanism in the script itself instead of allowing code owners to add their own .ignorefilesizes - then the barrier for adding large files is higher, but it remains possible.

sure